src/bindings/python/tests/test_graph/test_core.py

```diff
@@ -392,6 +392,25 @@ def test_discrete_type_info():
     assert n3.get_type_info().name < n1.get_type_info().name
     assert n3.get_type_info().name <= n1.get_type_info().name
 
+def test_shape_equality_with_list():
+    a = [1, 2, 3]
```

Comment:
Please, use [parametrization](https://docs.pytest.org/en/7.0.x/how-to/parametrize.html)